Sources/DiffableDataSource/DiffableTableViewDataSource+Replace.swift

```diff
@@ -10,30 +10,42 @@ import UIKit
 extension DiffableTableViewDataSource {
     // MARK: - Public API
 
-    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) throws {
-        try checkUniqueDiffableSections(sections: sections)
+    public func replace(with sections: [DiffableSection<SectionID, RowID>], animatingDifferences: Bool = true) {
+        switch checkUniqueDiffableSections(sections: sections) {
+        case let .success(sections):
+            self.unsafeReplace(sections: sections, animatingDifferences: animatingDifferences)
 
-        var snapshot = NSDiffableDataSourceSnapshot<SectionID, RowID>()
-
-        snapshot.appendSections(sections.map(\.identifier))
-
-        sections.forEach {
-            snapshot.appendItems($0.rows, toSection: $0.identifier)
+        case let .failure(error):
+            logFault(error: error)
         }
-
-        apply(snapshot, animatingDifferences: animatingDifferences)
     }
 
-    public func checkUniqueDiffableSections(sections: [DiffableSection<SectionID, RowID>]) throws {
+    public func checkUniqueDiffableSections(
+        sections: [DiffableSection<SectionID, RowID>]
+    ) -> Result<[DiffableSection<SectionID, RowID>], DiffableError> {
         let sectionIDs = sections.map { $0.identifier }
         let rowIDs = sections.flatMap { $0.rows }
 
-        guard sectionIDs.unique().count == sectionIDs.count else {
-            throw DiffableError.sectionsNotUnique
+        if sectionIDs.unique().count != sectionIDs.count {
+            return .failure(DiffableError.sectionsNotUnique)
+        } else if rowIDs.unique().count != rowIDs.count {
+            return .failure(DiffableError.rowsNotUnique)
+        } else {
+            return .success(sections)
         }
```

Comment:
[nit] easier to read perhaps; feel free to ignore.

```suggestion

        guard sectionIDs.unique().count == sectionIDs.count else {
            return .failure(DiffableError.sectionsNotUnique)
        } 
        
        guard rowIDs.unique().count == rowIDs.count else {
            return .failure(DiffableError.rowsNotUnique)
        } 
        
        return .success(sections)
    }
```